### DISCUSSION TITLE
unit of measure

### DISCUSSION THREAD

**Discussion by adamgreenberg07** [Hyundai-Kia-Connect/kia_uvo#797] [Q&A]
There is a comment in the installation readme that says "By default, distance unit is based on HA metric/imperial preference, you need to configure each entity if you would like other units."

How do I configure the entity to report in metric vs. imperial without changing my home assitant default?

I assume there is a line in the yaml code, but what exactly is it?

I've tried "unit_system: metric" (has no impact) and "unit_of_measurement: metric" (adds the word metric but does not change the number).

**Comment by cdnninja** [comment]
It exists in the guise of if you select the sensor and go into settings on it. Some power ones can't be changed.

**Reply by NOLBwithaD** [reply]
Hi @fuatakgun, 
This is a brilliant addition to HA.  I have been using it for over 2 years, but one thing has outfoxed me. 
I too would like to change the default measurements, in my case from Kilometres to miles, but I too cannot understand how to change it just for this app.  For example in the sensor.py script I see this:
SENSOR_DESCRIPTIONS: Final[tuple[SensorEntityDescription, ...]] = (
    SensorEntityDescription(
        key="_total_driving_range",
        name="Total Driving Range",
        icon="mdi:road-variant",
        device_class=SensorDeviceClass.DISTANCE,
        native_unit_of_measurement=DYNAMIC_UNIT,

I cannot understand what I might need change that would convert from miles to kilometres.  Am I even looking in the correct place?

I do hope that you can point myself, and @adamgreenberg07 in the right direction.
Many thanks.

**Reply by adamgreenberg07** [reply]
The cog in the upper right of the entity has the settings

![Screenshot_20240111-183705~2.png](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/67129417/b9335e8e-f175-46e8-bf12-727927a7ee13)

**Reply by NOLBwithaD** [reply]
Doh - what a clot!  Thank you @adamgreenberg07.  I can't believe I missed that.  Sometimes the bleedin' obvious is staring you right in the eyes.  So sorry to trouble you.  You are a first class star. and so quick.  Much appreciated!

**Reply by fuatakgun** [reply]
From a historical perspective, we had that option to configure units of measurement but removed it when home assistant core devs introduced this functionality.

Thanks to this integration I have learnt how complex the measuring system across the world :-)
